### ISSUE TITLE
0 value becomes NaN

### ISSUE THREAD

**Issue by paggers** [ericmjonas/pybm3d#6]
When applying pybm3d to an image that has a smooth black perimeter with strict 0 values, it returns the image with np.nan values at those pixels.

**Comment by timmeinhardt** [comment]
Thank you for your feedback! Could you provide me with an exemplary image to reproduce the issue?

**Comment by paggers** [comment]
<img width="239" alt="screen shot 2017-10-31 at 4 24 18 pm" src="https://user-images.githubusercontent.com/18237073/32253404-12d02124-be58-11e7-92ba-9e6d57e0a147.png">

In this MR image, anything outside the knee area is masked and equals 0. I set std = 1 and pass it to pybm3d, and this is the result.

**Comment by paggers** [comment]
![knee](https://user-images.githubusercontent.com/18237073/32253608-06bd195e-be59-11e7-85d8-43b515ac4ee6.jpg)
 
Original image

**Comment by timmeinhardt** [comment]
Thank you for the example image. I managed to reproduce equally bad results. The issue is caused by a "wrong" input format and data type. With the following preprocessing I got decent and non-corrupted results:

```
import numpy as np
from skimage import io
import pybm3d

img = io.imread('mr_img.jpg', as_grey=True)
img = 255.0 * img
img = img.astype(np.uint8)
out = pybm3d.bm3d.bm3d(img_grey, 1.0)
io.imsave('mr_img_out.jpg', out)
```

In general the input should always be of type integer and have either 1 or 3 color channels. We will improve the interface of the `pybm3d.bm3d.bm3d` function to account for these requirements. Thank you for your feedback!

I will leave this issue open until the interface is improved.

**Comment by timmeinhardt** [comment]
In commit https://github.com/ericmjonas/pybm3d/commit/70aa73dc2d685fd51f419bd4f0e112455b03f780 we introduce a change to the interface which raises an exception if an input image with a data type other than integer or a wrong number of channels is provided.

Unfortunately the underlying C core requires not-normalized float inputs. A way to securely provide these inputs is to require integer inputs (0 to 255) which are then converted to float. We refrained from including a float to integer (unnormalize) conversion (as described in my previous post) into our function as this might introduce inaccuracies.

@paggers Please close, if you feel your issue was addressed properly.

**Comment by paggers** [comment]
tl;dr 
Scale and shift data to fit [0,255] 
Convert to np.int8
